### ISSUE TITLE
Unaccessed parameter applies even when there is an inital _

### ISSUE THREAD

**Issue by coderanger** [microsoft/pyright#1931]
**Describe the bug**

An initial _ on function parameters has generally been used to mark them as intentionally unused. Pyright doesn't seem to reflect this.

**To Reproduce**

```python
def foo(_x):
    pass
```

**Expected behavior**

`_x` should not be reported as an unused symbol.

**Comment by erictraut** [comment]
An underscore within a function parameter already has a different meaning in the type checking world. It's the way of specifying that a parameter is position-only.

Pyright never generates an error or warning for an unaccessed parameter. It only emits informational hints that an editor like VS Code will use to apply a "grayed out" appearance.

**Comment by jakebailey** [comment]
FWIW, it's a double underscore parameter that signifies positional only, not a single underscore.

**Comment by coderanger** [comment]
> FWIW, it's a double underscore parameter that signifies positional only, not a single underscore.

```python
def foo(__x):
    pass
```

This also produces the same output. And while it's not a warning or error, it does show up in VSCode via Pylance which makes visual noise. If it needs to be a double underscore that's cool, but there should be some way to mark a parameter as intentionally unused.

**Comment by erictraut** [comment]
This is a duplicate of https://github.com/microsoft/pyright/issues/1836. I don't think anything has changed since we last discussed this topic that would change our minds.

Parameter names are meaningful in Python except in the case where they're position-only, so I wouldn't recommend to someone that they should change the name of a parameter just to eliminate the "grayed out" appearance.

There is no harm in a parameter being grayed out. It simply means that it's not accessed.

**Comment by coderanger** [comment]
The parameter color doesn't bother me but the line highlight and visible text aren't my favorite. Maybe your theme doesn't show these?

![Screen Shot 2021-06-01 at 4 00 41 PM](https://user-images.githubusercontent.com/128243/120400556-f26c7380-c2f2-11eb-8bb4-22489d293fa2.png)

**Comment by jakebailey** [comment]
That's coming from the Error Lens extension, which by default shows even informational diagnostics as text.

**Comment by jakebailey** [comment]
See: https://github.com/phindle/error-lens/issues/35

**Comment by erictraut** [comment]
Yeah, if another extension is treating this as a diagnostic rather than a hint, then it's a bug in that extension and should be fixed there.

**Comment by coderanger** [comment]
To summarize from the linked ticket, you can remove `hint` from `enabledDiagnosticLevels` in your settings. Maybe as a middle ground could also update the message to something like `Parameter "{}" is not accessed` which would allow for regex masking that more specifically without removing all hints or all unaccessed symbols?

**Comment by erictraut** [comment]
An informational diagnostic with a hint isn't meant to be displayed as text, so the message text should be irrelevant. If someone is displaying it as such, I would consider that a bug.

**Comment by coderanger** [comment]
A reasonable position but since not every extension will always share your views, saying "disable it or live with it" isn't the most helpful approach for something shipping as the default Python language server now. That's why Error Lens has more complex filtering options than just on vs off. Another option would be per-level rexeges on the Error Lens side.

**Comment by erictraut** [comment]
Adding a means for everyone to work around a bug in Error Lens probably isn't the right solution here. Working with the Error Lens extension author to fix the problem would lead to a much better result for everyone.
